### PR TITLE
Remove VerifySignatures Flag From ProposerSlashings and Transfers

### PR DESCRIPTION
**Pull request by 0xKiwi**
Part of #2951

This PR removes the `verifySignatures` flag from the ProposerSlashings and Transfers function so that they always have to involve signatures. The main code changes in this PR are updating the tests to always sign.

### CODE REVIEW

**Comment by terencechain** [comment]
The diff doesn't seem right

File: beacon-chain/core/blocks/block_operations.go [hunk 1]
```diff
@@ -209,15 +210,15 @@ func ProcessBlockHeader(
 //     )
 func ProcessRandao(
 	beaconState *pb.BeaconState,
-	body *pb.BeaconBlockBody,
+	body *ethpb.BeaconBlockBody,
 	verifySignatures bool,
```
Comment (terencechain) [inline beacon-chain/core/blocks/block_operations.go]: remove

File: beacon-chain/core/blocks/block_operations.go [hunk 2]
```diff
@@ -425,7 +420,7 @@ func slashableAttesterIndices(slashing *pb.AttesterSlashing) []uint64 {
 // appended to the BeaconState's latest attestations.
 func ProcessAttestations(
 	beaconState *pb.BeaconState,
-	body *pb.BeaconBlockBody,
+	body *ethpb.BeaconBlockBody,
 	verifySignatures bool,
```
Comment (terencechain) [inline beacon-chain/core/blocks/block_operations.go]: remove

File: beacon-chain/core/blocks/block_operations.go [hunk 3]
```diff
@@ -474,7 +469,7 @@ func ProcessAttestations(
 //    assert data.crosslink.parent_root == hash_tree_root(parent_crosslink)
 //    assert data.crosslink.data_root == Bytes32()  # [to be removed in phase 1]
 //    validate_indexed_attestation(state, convert_to_indexed(state, attestation))
-func ProcessAttestation(beaconState *pb.BeaconState, att *pb.Attestation, verifySignatures bool) (*pb.BeaconState, error) {
+func ProcessAttestation(beaconState *pb.BeaconState, att *ethpb.Attestation, verifySignatures bool) (*pb.BeaconState, error) {
```
Comment (terencechain) [inline beacon-chain/core/blocks/block_operations.go]: remove `verifySignatures`

**Comment by 0xKiwi** [comment]
I'm doing it in parts, removing all of them would be a lot of work to do in
one PR. I'd prefer to split it up.

**Comment by terencechain** [comment]
> I'm doing it in parts, removing all of them would be a lot of work to do in one PR. I'd prefer to split it up.
> […](#)

Ah sounds good. I misread the title. Apologize for that!
In that case this is ready to merge after we fix the diff

**Pull request merged**